Apps/WebClient/src/ClientApp/src/views/publicVaccineCard.vue

```diff
@@ -224,58 +246,41 @@ export default class PublicVaccineCardView extends Vue {
+        if (this.vaccineRecord !== undefined) {
+            const mimeType = this.vaccineRecord.document.mediaType;
+            const downloadLink = `data:${mimeType};base64,${this.vaccineRecord.document.data}`;
+            fetch(downloadLink).then((res) => {
+                SnowPlow.trackEvent({
+                    action: "download_card",
+                    text: "Public COVID Card PDF",
+                });
+                res.blob().then((blob) => {
+                    saveAs(blob, "VaccineProof.pdf");
+                });
+            });
+        }
+    }
+
     private downloadVaccinePdf() {
-        this.downloadError = null;
-        this.isDownloadingPdf = true;
-        this.vaccinationStatusService
-            .getPublicVaccineStatusPdf(
-                this.phn.replace(/ /g, ""),
-                this.dateOfBirth,
-                this.dateOfVaccine
-            )
-            .then((result) => {
-                if (result.resultStatus == ResultType.Success) {
-                    SnowPlow.trackEvent({
-                        action: "download_card",
-                        text: "Public COVID Card PDF",
-                    });
-                    const mimeType = result.resourcePayload.document.mediaType;
-                    const downloadLink = `data:${mimeType};base64,${result.resourcePayload.document.data}`;
+        this.retrievePublicVaccineRecord({
+            phn: this.phn.replace(/ /g, ""),
+            dateOfBirth: this.dateOfBirth,
+            dateOfVaccine: this.dateOfVaccine,
+        });
+    }
 
-                    fetch(downloadLink).then((res) => {
-                        res.blob().then((blob) => {
-                            saveAs(blob, "VaccineProof.pdf");
-                        });
-                    });
-                } else {
-                    this.logger.error(
-                        "Error returned when retrieving vaccine proof PDF: " +
-                            JSON.stringify(result.resultError)
-                    );
-                    this.downloadError = ErrorTranslator.toBannerError(
-                        "Retrieve PDF Error",
-                        result.resultError
-                    );
-                }
-            })
-            .catch((err) => {
-                this.logger.error(err);
-                this.downloadError = ErrorTranslator.toBannerError(
-                    "Retrieve PDF Error",
-                    err
-                );
-            })
-            .finally(() => {
-                this.isDownloadingPdf = false;
-            });
+    private get isLoadingVaccineRecordPdf(): boolean {
+        return this.vaccineRecordIsLoading || this.isDownloading;
     }
 }
 </script>
 
 <template>
     <div class="background flex-grow-1 d-flex flex-column">
         <loading
-            :is-loading="isLoading || isDownloading || isDownloadingPdf"
+            :is-loading="isLoading || isLoadingVaccineRecordPdf"
```

Comment:
If we're going to introduce a getter here, we should name it `isLoading`, rename the existing `isLoading` property to `isVaccinationStatusLoading`, and include it in the getter.